software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/MongoDBAuthenticationUtils.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.entity.nosql.mongodb;
+
+import org.apache.brooklyn.util.text.Strings;
+
+import org.apache.brooklyn.api.entity.EntitySpec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.brooklyn.api.entity.Entity;
+
+
+public class MongoDBAuthenticationUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MongoDBAuthenticationUtils.class);
+
+    private MongoDBAuthenticationUtils(){}
+
+    /**
+     * @return true if either the keyfile contents or keyfile url is set, false otherwise. If both are set, an IllegalStateException is thrown
+     */
+    public static boolean usesAuthentication(Entity entity) {
+        String keyfileContents = entity.config().get(MongoDBAuthenticationMixins.MONGODB_KEYFILE_CONTENTS);
+        String keyfileUrl = entity.config().get(MongoDBAuthenticationMixins.MONGODB_KEYFILE_URL);
+        if (!(Strings.isEmpty(keyfileContents) || Strings.isEmpty(keyfileUrl))) {
+            throw new IllegalStateException("keyfile contents and keyfile location cannot both be set");
+        }
+        if (Strings.isEmpty(keyfileContents) && Strings.isEmpty(keyfileUrl)) {
+            return false;
+        }
+        return true;
+    }
+
+    public static String getRootPassword(Entity entity) {
+        String password = entity.config().get(MongoDBAuthenticationMixins.ROOT_PASSWORD);
+        if (Strings.isEmpty(password)) {
+            LOG.debug(entity + " has no password specified for " + MongoDBAuthenticationMixins.ROOT_PASSWORD.getName() + "; using a random string");
+            password = Strings.makeRandomId(8);
+            entity.sensors().set(MongoDBAuthenticationMixins.ROOT_PASSWORD, password);
+            entity.config().set(MongoDBAuthenticationMixins.ROOT_PASSWORD, password);
+        }
+        return password;
+    }
+
+    public static void setAuthenticationConfig(EntitySpec<?> spec, Entity source) {
```

Comment:
The order of the arguments to these two methods surprised me. Worth documenting the effect of both or giving the arguments meaningful names.